edb/schema/types.py

```diff
@@ -521,22 +528,9 @@ def from_subtypes(cls, schema, subtypes, typemods=None, *, name=None):
             raise errors.UnsupportedFeatureError(
                 f'nested arrays are not supported')
 
-        if typemods:
-            dimensions = typemods[0]
-        else:
-            dimensions = []
-
-        if isinstance(stype, cls):
-            # There is no array of arrays, only multi-dimensional arrays.
-            element_type = stype.get_element_type(schema)
-            if not dimensions:
-                dimensions.append(-1)
-            dimensions += stype.dimensions
-        else:
-            element_type = stype
-            dimensions = []
+        dimensions = [-1]
```

Comment:
Please add a comment here, explaining that this means a 1 dimensional unbound array